### ISSUE TITLE
Add windows support

### ISSUE THREAD

**Issue by eborden** [haskell-CI/haskell-ci#199]
Travis has support for windows. Many packages are currently using appveyor, which works well, but requires a package to use two build systems. Adding windows support to this script would likely greatly increase the number of Haskell packages that test on windows.

https://docs.travis-ci.com/user/reference/windows/

**Comment by RyanGlScott** [comment]
Wow, I had no idea that Travis supported Windows until I saw this issue! That's a very intriguing development.

Out of curiosity, @eborden, are you currently using Travis + Windows for any of your projects (and could give anecdotes about how well it works in practice)? I ask since I had previously requested that `haskell-ci` support AppVeyor in #135, but if Travis accomplishes the same thing, then it might be less work in the long run to just go the Travis route.

**Comment by eborden** [comment]
@RyanGlScott I am not currently, but I plan to experiment with `network`. If that experiment proves fruitful then I'll look in to merging that work back here.

**Comment by hvr** [comment]
Fwiw, the approach should be fairly simlar to appveyor by using Chocolatey which iirc is supported by Travis/Windows as well

https://hub.zhox.com/posts/chocolatey-introduction/

**Comment by eborden** [comment]
@hvr yeah the hairy part of this that I haven't dug into yet is the ability to `case` on what shell you are in within the `install` etc stanzas.

**Comment by hvr** [comment]
@eborden yeah, that's also the part that worries me; the prospect of having to have severals commands conditional on windows vs osx/linux, ala

```yaml
- if [[ "$TRAVIS_OS_NAME" == "windows" ]]; then ...; fi
```

doesn't excite me too much ;)))

But it appears we have Bash scripting available by default on Windows too... that's something
